### DISCUSSION TITLE
Option to clear history (important for removing sensitive info / security)

### DISCUSSION THREAD

**Discussion by binaryfire** [outline/outline#8067] [Ideas]
We've have a couple of permissions-restricted documents where we temporarily included some credentials. We've now moved them into a secrets manager but can't remove the credentials from Outline because the previous versions of the docs are still viewable under "History".

It'd be great to have an option to clear / purge a document's history. It's essential from a compliance / security perspective. The only other alternative is to delete and recreate docs any time sensitive information changes.

**Comment by tommoor** [comment]
Interesting, yea – it makes sense as an option to have but I'm not sure quite what the model should be for allowing this – perhaps the revision record remains but the content is removed.

**Reply by binaryfire** [reply]
Yeah that would work. With some kind of message saying the history has been deleted. Could include the name of user who deleted the history and the time it was deleted too. 

You’d also want to limit the ability to delete history to admins only.

**Reply by mzeecedric** [reply]
Yes, definitely should be documented which user removed the history and ideally a reason why. Maybe only admins get permission to do that. I also expect that only single revisions can be removed, not everything.

**Reply by binaryfire** [reply]
Removing only a single revision wouldn’t solve the security issue I was referring to.

If a document which used to contain sensitive information has that information removed, all revisions need to be purged to completely remove that info from the system.

**Reply by mzeecedric** [reply]
> Removing only a single revision wouldn’t solve the security issue I was referring to.
> 
> If a document which used to contain sensitive information has that information removed, all revisions need to be purged to completely remove that info from the system.

@binaryfire, I think that usually there is one or maybe several versions that contain the sensitive information. If you have documents that have evolved over a long time you usually want to keep track of important changes and not purge everything. But this could be of course a feature to optionally select all revisions.
